src/impl/source/profiles/profileDiff.ts

```diff
@@ -0,0 +1,392 @@
+import { retrieveMetadata } from "../../../utils/retrieveMetadata";
+import { Org, LoggerLevel } from "@salesforce/core";
+import path from "path";
+import fs from "fs";
+import {
+  METADATA_INFO,
+  MetadataInfo
+} from "../../../impl/metadata/metadataInfo";
+import ProfileRetriever from "../../../impl/metadata/retriever/profileRetriever";
+import ProfileWriter from "../../../impl/metadata/writer/profileWriter";
+import Profile from "../../../impl/metadata/schema";
+import { SFPowerkit } from "../../../sfpowerkit";
+import cli from "cli-ux";
+import MetadataFiles from "../../../impl/metadata/metadataFiles";
+const jsdiff = require("diff");
+
+const CRLF_REGEX = /\r\n/;
+const LF_REGEX = /\n/;
+
+export default class ProfileDiffImpl {
+  private sourceOrg: Org = null;
+  public output = [];
+  private sourceLabel = "Local";
+  private targetLabel = "Remote";
+  public constructor(
+    private profileList: string[],
+    private sourceOrgStr: string,
+    private targetOrg: Org,
+    private outputFolder: string
+  ) {
+    this.targetLabel = this.targetOrg.getConnection().getUsername();
+  }
+  public async diff() {
+    SFPowerkit.log("Profile diff start. ", LoggerLevel.INFO);
+    let profileSource: Promise<any[]> = null;
+    //let profileXmlMapPromise: Promise<string[]> = null;
+    if (this.sourceOrgStr) {
+      SFPowerkit.log("Creating source org ", LoggerLevel.INFO);
+      this.sourceOrg = await Org.create({
+        aliasOrUsername: this.sourceOrgStr,
+        isDevHub: false
+      });
+    }
+    if (
+      (!this.profileList || this.profileList.length === 0) &&
+      this.sourceOrgStr
+    ) {
+      this.sourceLabel = this.sourceOrg.getConnection().getUsername();
+      SFPowerkit.log(
+        "Not profile provided, looding profile from source org. ",
```

Comment:
Spelling and let us update to loading all profiles from source org

Reply:
resolved